pennylane/ops/qubit/qchem_ops.py

```diff
@@ -597,18 +597,17 @@ def circuit(phi):
 
     def generator(self):
         w0, w1, w2, w3 = self.wires
-        coeffs = [0.0625, 0.0625, -0.0625, 0.0625, -0.0625, 0.0625, -0.0625, -0.0625]
-        obs = [
-            qml.X(w0) @ qml.X(w1) @ qml.X(w2) @ qml.Y(w3),
-            qml.X(w0) @ qml.X(w1) @ qml.Y(w2) @ qml.X(w3),
-            qml.X(w0) @ qml.Y(w1) @ qml.X(w2) @ qml.X(w3),
-            qml.X(w0) @ qml.Y(w1) @ qml.Y(w2) @ qml.Y(w3),
-            qml.Y(w0) @ qml.X(w1) @ qml.X(w2) @ qml.X(w3),
-            qml.Y(w0) @ qml.X(w1) @ qml.Y(w2) @ qml.Y(w3),
-            qml.Y(w0) @ qml.Y(w1) @ qml.X(w2) @ qml.Y(w3),
-            qml.Y(w0) @ qml.Y(w1) @ qml.Y(w2) @ qml.X(w3),
-        ]
-        return qml.Hamiltonian(coeffs, obs)
+        # coeffs = [0.0625, 0.0625, -0.0625, 0.0625, -0.0625, 0.0625, -0.0625, -0.0625]
+        return 0.0625 * (
+            qml.X(w0) @ qml.X(w1) @ qml.X(w2) @ qml.Y(w3)
```

Comment:
Is it intentional to use @ which changes behavior depending on new opmath being enabled or not? (alternative: qml.prod(..))

Reply:
Yep, it is intentional. So that when it is not enabled, we get a Hamiltonian.